### ISSUE TITLE
[Feature Request] Pass the Hash Support

### ISSUE THREAD

**Issue by IppSec** [ropnop/kerbrute#38]
It would be nice to have the ability to spray user accounts with NTLM Hashes.  The two main use cases I imagine for this is:
* Pulling an NTLM Hash off a local workstation, then spraying AD to look for password re-use
* Testing passwords from domain controller backups (old copies of NTDS.DIT).

Currently, this can be done with https://github.com/3gstudent/pyKerbrute, but is an extremely hacky solution using Python2.

**Comment by ropnop** [comment]
Good idea! This would require a bit of hack in gokrb5 but it can work. Basically instead of calling `client.NewWithPassword`, we'll call `client.NewWithKeytab` and manually create at temporary keytab with the NTLM hash as the encryption key. The "hack" will just be to change the library so it can accept a raw encryption key in RC4 instead of the expected plaintext password when calling [AddEntry](https://github.com/jcmturner/gokrb5/blob/663478bf457f1fc3275973bea5b7b787cd332015/v8/keytab/keytab.go#L128). Should have a workign branch in a few days - do you have a lab you could test in? My AD lab is down unfortunately I don't even have a working DC I can try kerbrute against at the moment

**Comment by P4cm4n90** [comment]
Any updates on this subject? I can help with testing.

**Comment by TryA9ain** [comment]
i can help too
